File: backend/src/InsuraTech.Application/Users/Commands/CreateAdvisor/CreateAdvisorCommand.cs
using InsuraTech.Application.Users.DTOs;
using MediatR;

namespace InsuraTech.Application.Users.Commands.CreateAdvisor;

public sealed record CreateAdvisorCommand : IRequest<UserResponse>
{
    public string   Email          { get; init; } = null!;
    public string   FirstName      { get; init; } = null!;
    public string   LastName       { get; init; } = null!;
    public string   Nationality    { get; init; } = null!;
    public string   BirthDate      { get; init; } = null!;
    public int      YearsInCompany { get; init; }
    public string   OfficeLocation { get; init; } = null!;
    public string   WorkSchedule   { get; init; } = null!;

    /// <summary>Id del líder autenticado — se inyecta en el controller desde el JWT.</summary>
    public Guid LeaderId { get; init; }
}

File: backend_v2/src/main/java/com/insuratech/application/users/commands/CreateAdvisorCommand.java
package com.insuratech.application.users.commands;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
public record CreateAdvisorCommand(
    @NotBlank @Email String email, @NotBlank String password,
    @NotBlank String firstName, @NotBlank String lastName,
    String phone, String city, String createdByLeaderId
) {}
